src/github_utils.py

```diff
@@ -0,0 +1,132 @@
+import json
+import os
+import requests
+from github import Github
+
+def get_pull_request_diff():
+    """Fetch the pull request diff content from GitHub"""
+    g = Github(os.getenv('GIT_TOKEN'))
+
+    # Parse GitHub event context
+    event_path = os.getenv('GITHUB_EVENT_PATH')
+    with open(event_path, 'r') as f:
+        event_data = json.load(f)
+
+    repo = g.get_repo(event_data['repository']['full_name'])
+    pr_number = event_data['pull_request']['number']
+    pull_request = repo.get_pull(pr_number)
+
+    # Get diff content
+    diff_url = pull_request.diff_url
+    response = requests.get(diff_url)
+    return response.text
+
+
+def get_valid_lines(diff_content):
+    """Parse diff to find valid line numbers with proper initialization"""
+    valid_lines = {}
+    current_file = None
+    line_number = None
+    
+    for line in diff_content.split('\n'):
```

Comment:
The `line_number = None` in the `if line.startswith('diff --git')` block seems redundant since it's reset again in the `if line.startswith('@@')` block.  Consider only setting it once.

```suggestion
Remove `line_number = None` in the `if line.startswith('diff --git')` block.
```